GameServer/gameobjects/GamePlayer.cs

```diff
@@ -6307,15 +6307,22 @@ protected override AttackData MakeAttack(GameObject target, InventoryItem weapon
 					{
 						(weapon as GameInventoryItem).OnStrikeTarget(this, target);
 					}
-					//Camouflage
-					if (target is GamePlayer && HasAbility(Abilities.Camouflage))
+					//Camouflage - Camouflage will be disabled only when attacking a GamePlayer or ControlledNPC of a GamePlayer.
+					if (HasAbility(Abilities.Camouflage) && target is GamePlayer || (target is GameNPC && (target as GameNPC).Brain is IControlledBrain && ((target as GameNPC).Brain as IControlledBrain).GetPlayerOwner() != null))
 					{
 						CamouflageEffect camouflage = EffectList.GetOfType<CamouflageEffect>();
-						if (camouflage != null)
+												
+						if (camouflage != null) // Check if Camo is active, cancel ability and put on disable timer.
 						{
 							DisableSkill(SkillBase.GetAbility(Abilities.Camouflage), CamouflageSpecHandler.DISABLE_DURATION);
 							camouflage.Cancel(false);
 						}
+						
+						Skill camo = SkillBase.GetAbility(Abilities.Camouflage);
+						int camoDisable = GetSkillDisabledDuration(camo);
+						
+						if (camoDisable <= 0) // If camo is not active and not on disable timer, disable it.
+							DisableSkill(camo, CamouflageSpecHandler.DISABLE_DURATION);

```

Comment:
aye, you are correct about a dupe. I can remove the first disableskill call, as conditions will still be met at the second one

Because yes, if ability is active, then 'camoDisable' will always be '0' , so 'if (camoDisable <= 0)' is true and skill is disabled. If player has ability, and ability is not active or on 'cooldown', then it will be disabled too. If ability is already 'cooldown' from a pervious attack, then do nothing.

I just realised i'm using the term 'cooldown' from League of Legends hehe, but I think you get it ;)
I'll push the change